File: src/Cryptofolio.Core/Asset.cs
namespace Cryptofolio.Core
{
    /// <summary>
    /// Models an asset.
    /// </summary>
    public class Asset
    {
        /// <summary>
        /// The id.
        /// </summary>
        public string Id { get; set; }

        /// <summary>
        /// The symbol.
        /// </summary>
        public string Symbol { get; set; }

        /// <summary>
        /// The name.
        /// </summary>
        public string Name { get; set; }

        /// <summary>
        /// The description.
        /// </summary>
        public string Description { get; set; }
    }
}

File: src/Cryptofolio.App/ClientApp/src/app/models/asset.ts
import { Currency } from "./currency";

export class Asset {
  id: string;
  symbol: string;
  name: string;
  description: string;
  thumbImageUrl: string;
  smallImageUrl: string;
  largeImageUrl: string;

  constructor(properties?: any) {
    Object.assign(this, properties);
    this.thumbImageUrl = properties["thumb_image_url"];
    this.smallImageUrl = properties["small_image_url"];
    this.largeImageUrl = properties["large_image_url"];
  }
}

export class AssetTicker {
  asset: Asset;
  vsCurrency: Currency;
  timestamp: Date;
  value: number;

  constructor(properties?: any) {
    Object.assign(this, properties);
    this.asset = new Asset(properties.asset);
    this.vsCurrency = new Currency(properties["vs_currency"]);
    this.timestamp = new Date(properties.timestamp);
  }
}